sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -839,7 +845,15 @@ private void validateReplicaAddresses(AddressInformation[] addresses) {
                 Arrays
                     .stream(addresses)
                     .map(address -> address.getPhysicalUri())
-                    .filter(addressUri -> addressUri.getHealthStatus() == Uri.HealthStatus.UnhealthyPending)
+                    .filter(addressUri -> this.replicaValidationScopes.contains(addressUri.getHealthStatus()))
+                    .sorted(new Comparator<Uri>() {
+                        @Override
+                        public int compare(Uri o1, Uri o2) {
```

Comment:
Can we please have unit tests for this logic?

Reply:
Yes, there is a test in GatewayAddressCacheTest to validate the sequence is unhealthPending, then unkown